### ISSUE TITLE
Refactor Leo's key handling code

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#1269]
**Status**
- [x] Remove evil kwargs.
- [x] Remove unused ivars.
- [x] Simplify and refactor the code:
  - **c.doCommandByName** looks up the proper command function.
    k.simulateCommand is now a thin wrapper around c.doCommandByName.
  - **c.insertCharFromEvent** executes the proper command function for any key,
    regardless of command bindings.
    It handles many internal details that were formerly in different methods.
- [x] Changed the default bindings of <tab> in leoSettings.leo.
- [x] Pass user testing for at least several weeks.
- [x] Merge keys branch into devel.

#1594 was originally intended to be part of this project. It is not likely ever to happen.

**What I did**

Major changes:

- c.doCommand no longer sets k.retVal.
- Removed k.commandExists: It was supposed to handle stuff after command names.
- Removed all dynamicMenu kwargs.
- Removed k.masterCommand!

Other changes:

- Eliminated k.newMinibufferWidget
- k.endCommand ignores its arguments and sets k.commandName = None
- Replace c.executeAnyCommand by try/except block.
- Remove unused k.argSelectedText ivar.
- Remove unused k.newMinibufferWidget ivar.
- Remove 'specialCallback' logic.
- Remove mouseClick kwarg from k.keyboardQuit.

Files changed:

- leoAtFile.py
- leoCommands.py, commanderEditCommands.py, controlCommands.py, editCommands.py
- leoKeys.py, leoMenu.py, leoVim.py
- qt_gui.py, qt_text.py, qt_tree.py

**Compensating for a missing PR**

It would have been best to issue a PR. The PR would have shown all and *only* the changes made in the keys branch.  The following Leonine script shows those changes, and many more besides:
```python
import leo.commands.editFileCommands as efc

efc.GitDiffController(c).diff_two_revs(
    rev1='4b8320485c0', # first keys commit
    rev2='2507a1fcc70', # Latest devel commit
    directory=None,
)
```

**Comment by edreamleo** [comment]
I am reopening this. This code requires changed bindings for tab. These changes were reverted when merging the keys branch into devel.

**Comment by edreamleo** [comment]
Fixed at rev 64c49da in devel. Only one binding changed:

old: `newline-and-indent = !text = Tab`
new: `indent-region = !text = Tab`
